File: rust/src/bin/03-fs.rs
/*
 * This tutorial shows basic file system operations in Zbox.
 */

extern crate zbox;

use zbox::{init_env, RepoOpener};

fn main() {
    // initialise zbox environment, called first
    init_env();

    // create and open a repository
    let mut repo = RepoOpener::new()
        .create(true)
        .open("[your_repo_uri]", "secret password")
        .unwrap();

    // get repo metadata
    let info = repo.info().unwrap();
    println!("{:?}", info);

    // create test file and dir
    repo.create_file("/file").unwrap();
    repo.create_dir("/dir").unwrap();

    // check if path exists
    let result = repo.path_exists("/dir").unwrap();
    println!("{}", result);

    // get metadata of a path
    let metadata = repo.metadata("/dir").unwrap();
    println!("{:?}", metadata);

    // copy file
    repo.copy("/file", "/file_copy").unwrap();

    // rename file
    repo.rename("/file", "/file_new").unwrap();

    // remove test file and dir
    repo.remove_file("/file_new").unwrap();
    repo.remove_file("/file_copy").unwrap();
    repo.remove_dir("/dir").unwrap();
}

File: nodejs/03-fs.js
/*
 * This tutorial shows basic file system operations in Zbox.
 */

const Zbox = require('@zbox/nodejs');

(async function run() {
  // create a Zbox instance
  const zbox = new Zbox();

  // initialise Zbox environment and turn on debug logs
  await zbox.initEnv({ log: { level: 'debug' } });

  // open the repo
  var repo = await zbox.openRepo({
    uri: '[your_repo_uri]',
    pwd: 'secret password',
    opts: { create: true }
  });

  // get repo metadata
  let info = await repo.info();
  console.log(info);

  // create test file and dir
  let file = await repo.createFile('/file');
  await file.close();
  await repo.createDir('/dir');

  // check if path exists
  let result = await repo.pathExists('/dir');
  console.log(result);

  // get metadata of a path
  let metadata = await repo.metadata('/dir');
  console.log(metadata);

  // copy file
  await repo.copy({ from: '/file', to: '/file_copy' });

  // rename file
  await repo.rename({ from: '/file', to: '/file_new' });

  // remove test file and dir
  await repo.removeFile('/file_new');
  await repo.removeFile('/file_copy');
  await repo.removeDir('/dir');

  // close repo and exit Zbox
  await repo.close();
  await zbox.exit();
})();
